enhancements/results-forwarding.md

```diff
@@ -0,0 +1,504 @@
+
+Required: False
+Type: string
+
+Reference to the secret containing the token.
+
+#### `resultForwarding.grpc.extraMetadata.clusterName`
+
+Required: True
+Type: string
+
+The name of the cluster to use in result payloads.
+
+#### `resultForwarding.grpc.extraMetadata.clusterType`
+
+Required: True
+Type: string
+
+The type of the cluster to use in result payloads.
+
+#### `resultForwarding.grpc.extraMetadata.randomKey`
+
+Required: False
+Type: string
+
+This is simply and additional way to pass in data to the forwarding
+implementation.
+
+#### `ScanSetting` Examples
+
+For reference, the following `ScanSetting` is available by default and
+supported today with `v1alpha1` `ScanSetting` CRDs.
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce 
+  rotation: 3 
+  size: 1Gi 
+roles:
+- worker 
+- master 
+scanTolerations: 
+  default:
+  - operator: Exists
+  schedule: 0 1 * * * 
+```
+
+To forward results to a gRPC endpoint using mutual TLS:
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce 
+  rotation: 3 
+  size: 1Gi 
+resultForwarding:
+  provider: grpc
+  grpc:
+    version: v1
+    endpoint: https://compliance-endpoint.example.com
+    tls:
+      dynamic: 
+        caSecretName: $caSecretName
+        serverSecretName: $ref
+        clientSecretName: $ref
+    authentication:
+      type: mtls
+    extraMetadata:
+      clusterName: foo
+      clusterType: bar
+roles:
+- worker 
+- master 
+scanTolerations: 
+  default:
+  - operator: Exists
+  schedule: 0 1 * * * 
+```
+
+The following forwards results to a gRPC endpoint using token authentication.
+It also sets additional metadata for the gRPC implementation.
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce 
+  rotation: 3 
+  size: 1Gi 
+resultForwarding:
+  provider: grpc
+  grpc:
+    version: v1
+    endpoint: https://compliance-endpoint.example.com
+    tls:
+      dynamic: 
+        caSecretName: $caSecretName
+    authentication:
+      type: token
+      token:
+        tokenSecretName: $ref
+    extraMetadata:
+      clusterName: foo
+      clusterType: bar
+      randomKey: random value
+roles:
+- worker 
+- master 
+scanTolerations: 
+  default:
+  - operator: Exists
+  schedule: 0 1 * * * 
+```
+
+#### gRPC API
+
+This section focuses solely on the details for implementing a forwarder for
+gRPC.
+
+```protobuf
+message Result {
+  string subject = 1;
+  string control = 2;
+  string rule = 3;
+  string outcome = 4;
+  string description = 5;
+  string severity = 6;
+  string instructions = 7;
```

Comment:
Should this also include the scan ID? Whatever is consuming this won't be able to group results by scan with the current proposal.

Thoughts?

Reply:
What data type would we have to use to make the ID usable for grouping? If it could be a random string, we could send along e.g. the scan UUID.

In general I agree we should be able to tell where the result comes from.

Reply:
I think using the scan UUID makes sense.